apiary.apib

```diff
@@ -0,0 +1,397 @@
+FORMAT: 1A
+HOST: http://deploy.cwp.govt.nz/naut
+
+# Deploynaut API
+
+This API is a newly developed interface for the CWP Dashboard. Data structure is based on the
+[JSON API specifications](http://jsonapi.org/).
+
+
+# Using the API [/]
+
+In order to use the API, you will need to acquire a token via Service Desk request. Any request you make to
+an endpoint should include the following:
+
+## Making a request: Headers [GET]
+
+These headers will authenticate you against Deploynaut to evaluate whether you have the permissions necessary
+to access the information you are requesting. You can send a GET request to www.deploy.cwp.govt.nz/naut to
+test that you are authenticating correctly - it will return your user information.
+
++ Request (application/json)
+
+    + Headers
+
+            X-api-version: 1.0
+            X-user-email: sherbert.lemon@hogwarts.uk
+            X-user-token: fa88787fcd99787
+            Accept: application/json
+            Host: deploy.cwp.govt.nz
+
+
++ Response 200 (application/json)
+
+        {
+            "data": [
+                {
+                    "type": "users",
+                    "id": "2379",
+                    "name": "Albus Dumbledore",
+                    "email": "sherbert.lemon@hogwarts.uk",
+                    "phone": "0203040506"
+                }
+            ]
+        }
+
+
+# Stacks [/projects]
+
+## List all stacks [GET /projects]
+
+Will return a list of all stacks that you have the authority to view, with relevant datapoints.
+
++ Response 200 (application/json)
+
+        {
+            "count": 1,
+            "data": [
+                {
+                    "type": "stacks",
+                    "id": "CWP001",
+                    "name": "CWP Stack 001",
+                    "primary_domain": "mom.govt.nz",
+                    "agency": "Ministry of Magic",
+                    "size": "large",
+                    "disaster_recovery": "active",
```

Comment:
Yeah that's interesting - and ties in with the Agency thing above. I feel like it might be important information to have (as the story says "information like the instance page") - but that might need to be an extension specific to CWP.

I can see having a `updateAPIObject()` hook where a CWPAPIExtension can exists and lump in whatever extra data it wants - and I think we'd need to get it off of Portal somehow. We'll see what Chris says with regards to the information he actually needs.